clang/lib/Sema/SemaAvailability.cpp

```diff
@@ -182,6 +182,12 @@ static bool ShouldDiagnoseAvailabilityInContext(
       return false;
   }
 
+  if (K == AR_Deprecated) {
+    if (const auto *VD = dyn_cast<VarDecl>(OffendingDecl))
+      if (VD->isLocalVarDecl() && VD->isDeprecated())
```

Comment:
```suggestion
      if (VD->isLocalVarDeclOrParm() && VD->isDeprecated())
```
Otherwise, this won’t handle cases such as:
```c++
[[deprecated]] void f([[deprecated]] int x) { x = 4; }
```
We should also add a test for that.